src/Octoshift/SecretScanningAlertService.cs

```diff
@@ -0,0 +1,163 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Octoshift.Models;
+using OctoshiftCLI;
+
+namespace Octoshift
+{
+    public class SecretScanningAlertService
+    {
+        private readonly GithubApi _sourceGithubApi;
+        private readonly GithubApi _targetGithubApi;
+        private readonly OctoLogger _log;
+
+        public SecretScanningAlertService(GithubApi sourceGithubApi, GithubApi targetGithubApi, OctoLogger logger)
+        {
+            _sourceGithubApi = sourceGithubApi;
+            _targetGithubApi = targetGithubApi;
+            _log = logger;
+        }
+
+        public virtual async Task MigrateSecretScanningAlerts(string sourceOrg, string sourceRepo, string targetOrg,
+            string targetRepo, bool dryRun)
+        {
+            _log.LogInformation($"Migrating Secret Scanning Alerts from '{sourceOrg}/{sourceRepo}' to '{targetOrg}/{targetRepo}'");
+
+            var sourceAlerts = await GetAlertsWithLocations(_sourceGithubApi, sourceOrg, sourceRepo);
+            var targetAlerts = await GetAlertsWithLocations(_targetGithubApi, targetOrg, targetRepo);
+            
+            _log.LogInformation($"Source {sourceOrg}/{sourceRepo} secret alerts found: {sourceAlerts.Count}");
+            _log.LogInformation($"Target {targetOrg}/{targetRepo} secret alerts found: {targetAlerts.Count}");
+            
+            _log.LogInformation("Matching secret resolutions from source to target repository");
+            foreach (var alert in sourceAlerts)
+            {
+                _log.LogInformation($"Processing source secret {alert.Alert.Number}");
+                
+                if (alert.Alert.State == "resolved")
```

Comment:
Here also I would create `const`s instead of using magic strings. For these constants you might also want to follow what we did for [RepositoryMigrationStatus](https://github.com/github/gh-gei/blob/8736d28aef48fb4cd7d784e1f7328d1b27470883/src/Octoshift/RepositoryMigrationStatus.cs#L3) if you intend to use them in other places.